File: app/src/main/java/com/katdmy/android/unocount/ScoreViewModel.java
package com.katdmy.android.unocount;

import android.app.Application;

import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;

import java.util.List;

public class ScoreViewModel extends AndroidViewModel {

    private ScoreRepository mRepository;
    private LiveData<ScoreMatrixCursor> mScoreMatrixCursor;
    private int mPlayerCount = 1;

    public ScoreViewModel(Application application) {
        super(application);
        mRepository = new ScoreRepository(mPlayerCount);
        mScoreMatrixCursor = mRepository.getCursor();
    }

    LiveData<ScoreMatrixCursor> getCursor() {
        return mScoreMatrixCursor;
    }

    void deleteAll() {
        mRepository.deleteAll();
    }

    void insert(List<Integer> currentScore) {
        mRepository.insert(currentScore);
    }

    void setPlayerCount(int playerCount) {
        mPlayerCount = playerCount;
        mRepository.setPlayerCount(playerCount);
    }
}

File: app/src/main/java/com/katdmy/android/unocount/ui/score/ScoreViewModel.kt
package com.katdmy.android.unocount.ui.score

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.Transformations
import androidx.lifecycle.ViewModel
import com.katdmy.android.unocount.domain.score.ScoreRepository
import com.katdmy.android.unocount.domain.score.model.Score

class ScoreViewModel(
        private val scoreRepository: ScoreRepository
) : ViewModel() {

    var players: LiveData<Array<String>> = Transformations.distinctUntilChanged(scoreRepository.players)
    var score: LiveData<List<Score>> = scoreRepository.score
    var totalScore: LiveData<IntArray> = Transformations.switchMap(score) { score -> getTotalFromScore(score) }

    private fun getTotalFromScore(score: List<Score>): LiveData<IntArray> {
        val size = players.value?.size ?: 0
        val total = IntArray(size) { 0 }
        if (size > 0)
            for (roundScore in score) {
                val arrayOfScores = roundScore.data.split(",")
                for (i in players.value?.indices!!) {
                    total[i] = total[i] + Integer.parseInt(arrayOfScores[i])
                }
            }
        return MutableLiveData(total)
    }

    fun deleteScore() = scoreRepository.deleteScore()
    fun insertPlayers(playersList: Array<String>) = scoreRepository.insertPlayers(playersList)
    fun insertScore(currentScore: List<Int>) = scoreRepository.insertScore(currentScore)

    init {
        players = scoreRepository.players
        score = scoreRepository.score
    }
}